src/coreclr/vm/loaderallocator.cpp

```diff
@@ -920,7 +920,7 @@ OBJECTREF LoaderAllocator::CompareExchangeValueInHandle(LOADERHANDLE handle, OBJ
     {
         OBJECTREF *ptr = (OBJECTREF *)(((UINT_PTR)handle) - 1);
 
-        gc.previous = InterlockedCompareExchangeT(ptr, gc.value, gc.compare);
+        gc.previous = ObjectToOBJECTREF(InterlockedCompareExchangeT((Object **)ptr, OBJECTREFToObject(gc.value), OBJECTREFToObject(gc.compare)));
```

Comment:
Would it be better to have a specific override for `InterlockedCompareExchangeT` that takes `OBJECTREF`? I didn't find any other uses of the API with `OBJECTREF` or `Object*`, so perhaps there is another way to perform this operation.

Reply:
The problem is that `InterlockedCompareExchangeT` internally calls `T` constructors. `OBJECTREF`s have to constructed and copied in particular way so that they remain protected. `InterlockedCompareExchangeT` violates the rules.

GC stress infrastructure checks for the common violation of these rules (look for `Thread::ObjectRefNew` and friends) and that's what causing the GC stress failure.

Reply:
This change is suppressing the OBJECTREF checking inside `InterlockedCompareExchangeT` by using raw `Object*`.